### PR TITLE
Fixes #3326. Menu is responding even outside the limits, not moving to the view that has the mouse.

### LINKED ISSUE #3326: Menu is responding even outside the limits, not moving to the view that has the mouse.
See on this video where the menu item is moved even outside it's limit and clicking below on the `ListView` or the `TableView` doesn't select any.

https://github.com/gui-cs/Terminal.Gui/assets/13117724/7a102a46-5849-44e8-9249-d84cb3420b80

### PR DESCRIPTION
**Pull request by BDisp**
## Fixes

- Fixes #3326

## Proposed Changes/Todos

- [x] Ensure the Menu use the correct view passed on the event
- [x] Add more unit tests to Menu and ContextMenu to proof

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [ ] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

File: UnitTests/Views/MenuBarTests.cs [hunk 1]
```diff
@@ -3658,4 +3658,26 @@ public CustomWindow ()
             Add (menu);
         }
     }
+
+    [Fact]
+    [AutoInitShutdown]
+    public void Click_Another_View_Close_A_Open_Menu ()
```
Comment (dodexahedron) [inline UnitTests/Views/MenuBarTests.cs:3664]: Minor name nitpick.

Should be "An" instead of "A".

**Review by dodexahedron — APPROVED** [review]
LGTM

Although could be less verbose using target-typed new instead of explicitly typed constructors, but that's not important.

**Comment by BDisp** [comment]
> LGTM
> 
> Although could be less verbose using target-typed new instead of explicitly typed constructors, but that's not important.

What do you mean specifically and where?

**Comment by BDisp** [comment]
I think the `Menu` don't need to subscribe the `Application.MouseEvent` event because when a menu is opened it grab the mouse and then the `MouseGrabView` can be used to handle it, as before. It's only a note and leaving as it's.

**Review by tig — APPROVED** [review]

**Pull request merged**